File: Java_program/Problem_on_Numbers/ArmstrongNumber.java
//////////////////////////////////////////////////////////////////////////////////////////////
//
//required header file
//
//////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;                   // for taking input from user
import java.lang.Math;                      //for exponent use

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  Class Name  :   Logic
//  Function Name : armstrong
//  Description :   checks armstrong number
//  Input :         Integer
//  output :        Integer
//  Author :        Rekha Shankarlal Kumawat
//  Date :          10/12/2025
//
//////////////////////////////////////////////////////////////////////////////////////////////

class Logic
{
    int iNo ;
    public Logic(int iNo)
    {
        this.iNo = iNo;
    }

    public int armstrong()
    {
        int iDigit = 0 ;
        int iSum = 0 ;
        int iCount = 0 ;
        int iTemp = iNo ;

        while(iNo != 0 )
        {
            
            iDigit = iNo % 10 ;
            iNo = iNo /10 ;
            iCount++;
        }

        while(iTemp != 0 )
        {
            iDigit = iTemp % 10 ;
            iTemp = iTemp /10 ;
            iSum = iSum + (int)Math.pow(iDigit, iCount);
        }
       return iSum ;
    }
}// end of Logic Class

//////////////////////////////////////////////////////////////////////////////////////////////
//
//Entry point class
//
//////////////////////////////////////////////////////////////////////////////////////////////

class ArmstrongNumber
{
    public static void main(String Arr [])
    {
        Scanner sobj = new Scanner(System.in);
        int iRet = 0 ;
        int iValue = 0 ;
        System.out.println("Enter the number:");
        iValue = sobj.nextInt();

        Logic lobj = new Logic(iValue);
        iRet = lobj.armstrong();

        if(iRet == iValue)
        {
            System.out.println(iValue+ " is an armstrong number");
        }
        else
        {
            System.out.println(iValue+ " is not an armstrong number");
        }
    }
}//end of main function

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  Input1:153                        Output:153 is an armstrong number
//  Input1:123                        Output:123 is not an armstrong number
//
//////////////////////////////////////////////////////////////////////////////////////////////

File: Python_Programming/Problems_on_Numbers/ArmStrongNumber.py
'''
    Description : program  to check the number is Armstrong number or not 
                EX :- 153 (1^3 +5^3 + 3^3 == 153)

    Date :- 21/03/2026
    
    Author : Rekha Shankarlal Kumawat
'''

def ArmstrongNumber(iNo):

        iCount = 0 
        iDigit = 0 
        iPow = 0  
        iTemp = 0 
        iSum = 0  
        iCountTemp =0 
        
        iTemp = iNo

        while(iNo != 0):
        
            iDigit = iNo % 10               
            iCount = iCount +1
            iNo = iNo // 10
        
        iNo = iTemp 
        iCountTemp = iCount 

        while (iNo != 0):
            iDigit = iNo % 10
            iCount= iCountTemp
            iPow =1 

            while(iCount!= 0):
        
                iPow = iPow*iDigit
                iCount = iCount -1
            

            iSum =iSum +iPow 
            iNo = iNo // 10

        return iSum == iTemp

def main():

    iValue = 0
    iRet = False

    print("Enter the Numner : ")
    iValue = int(input())

    iRet = ArmstrongNumber(iValue)

    if(iRet == True):
        print(iValue," is a ArmStrong Number ")
    else:
        print(iValue ," is not a ArmStrong Number ")

if __name__ == "__main__":
    main()